hoot-core/src/main/cpp/hoot/core/io/OsmCsvWriter.cpp

```diff
@@ -0,0 +1,291 @@
+/*
+ * This file is part of Hootenanny.
+ *
+ * Hootenanny is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * --------------------------------------------------------------------
+ *
+ * The following copyright notices are generated automatically. If you
+ * have a new notice to add, please use the format:
+ * " * @copyright Copyright ..."
+ * This will properly maintain the copyright information. DigitalGlobe
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2018 DigitalGlobe (http://www.digitalglobe.com/)
+ */
+
+#include "OsmCsvWriter.h"
+
+#include <hoot/core/util/ConfigOptions.h>
+#include <hoot/core/util/Factory.h>
+#include <hoot/core/util/OsmUtils.h>
+
+#include <QFileInfo>
+
+using namespace std;
+
+namespace hoot
+{
+
+HOOT_FACTORY_REGISTER(OsmMapWriter, OsmCsvWriter)
+
+OsmCsvWriter::OsmCsvWriter()
+  : _precision(ConfigOptions().getWriterPrecision()),
+    _separator(","),
+    _endl("\n")
+{
+}
+
+OsmCsvWriter::~OsmCsvWriter()
+{
+}
+
+void OsmCsvWriter::open(QString url)
+{
+  QFileInfo path(url);
+  QString base = QString("%1/%2").arg(path.absolutePath()).arg(path.baseName());
+  //  Create new filenames, i.e. /path/filename.csv turns into /path/filename-nodes.csv
+  array<QString, FileType::MaxFileType> filenames;
+  filenames[FileType::Nodes] = base + QString("-nodes.") + path.completeSuffix();
+  filenames[FileType::Ways] = base + QString("-ways.") + path.completeSuffix();
+  filenames[FileType::Relations] = base + QString("-relations.") + path.completeSuffix();
+  filenames[FileType::WayNodes] = base + QString("-waynodes.") + path.completeSuffix();
+  filenames[FileType::RelationMembers] = base + QString("-relationmembers.") + path.completeSuffix();
```

Comment:
Each output file is an extension of the original output file.  Update the filenames to append `-nodes` for the node file.